imagepipeline/src/main/java/com/facebook/imagepipeline/cache/BufferedDiskCache.java

```diff
@@ -125,6 +125,32 @@ public Boolean call() throws Exception {
   }
 
   /**
+   * Performs disk cache check synchronously.
+   * @param key
+   * @return true if the key is found in disk cache else false
+   */
+  public boolean syncDiskCheck(final CacheKey key){
+    if(containsSync(key)){
+      return true;
+    }
+    EncodedImage result = mStagingArea.get(key);
```

Comment:
This looks exactly like the task content inside `containsAsync(CacheKey)` so it be great if you could change that to call this method rather than having the duplication.